awpy/visualization/plot.py

```diff
@@ -326,3 +341,382 @@ def plot_nades(
     axes.get_xaxis().set_visible(b=False)
     axes.get_yaxis().set_visible(b=False)
     return figure, axes
+
+
+def _plot_map_control_snapshot_helper(
+    map_name: str,
+    ct_tiles: dict,
+    t_tiles: dict,
+    axes: plt.Axes,
+    player_data: dict | None = None,
+) -> None:
+    """Helper function to plot map control nav tile plot.
+
+    Args:
+        map_name (str): Map used position_transform call
+        ct_tiles (dict): CT map control values dictionary
+        t_tiles (dict): T map control values dictionary
+        axes (plt.axes): axes object for plotting
+        player_data (dict): Dictionary of player positions
+            for each team. Expected format same as output of extract_player_positions
+
+    Returns: Nothing, all plotting is done on ax object
+
+    Raises:
+        ValueError: If map_name is not in awpy.data.NAV
+                    If player_dict not of expected
+                    extract_player_positions output format
+    """
+    if map_name not in NAV:
+        msg = "Map not found."
+        raise ValueError(msg)
+
+    ct_tile_set, t_tile_set = set(ct_tiles.keys()), set(t_tiles.keys())
+    all_tiles = ct_tile_set.union(t_tile_set)
+
+    # Iterate through the tiles that have a value
+    for tile in all_tiles:
+        if tile in NAV[map_name]:
+            area = NAV[map_name][tile]
+            se_x = position_transform(map_name, area["southEastX"], "x")
+            nw_x = position_transform(map_name, area["northWestX"], "x")
+            se_y = position_transform(map_name, area["southEastY"], "y")
+            nw_y = position_transform(map_name, area["northWestY"], "y")
+            width = se_x - nw_x
+            height = nw_y - se_y
+
+            # Use max value (if exists) for each side for the current tile
+            ct_val = max(ct_tiles[tile] + [0])
+            t_val = max(t_tiles[tile] + [0])
+
+            # Use each side's value above as weights for weighted average
+            # to find correct color
+            cur_color = min(ct_val / (ct_val + t_val), ct_val) * np.array(
+                [0, 1, 0]
+            ) + min(t_val / (ct_val + t_val), t_val) * np.array([1, 0, 0])
+
```

Comment:
Added a few lines documenting how the colors are calculated. Basically if a tile is just controlled by 1 team, then I want just a pure Green or Red. However, if there are tile values from both teams, then I take the sum of both teams' values. Then normalize the values and calculate the correct color.

Hm I might have messed up this color logic a bit now I think about it. I think I previously was mapping the tile values to the RGB color slightly different and this might be an artifiact of that old logic (basically previously I was using max() of tile values for each team and then calculating the color -> now I am using sum() of tile values for each team then calculating the color). I think just changing it to the below will fix it; however, I'll test this out with the notebook and make sure the plots still look okay. Thanks for catching this - def slipped by me.

`cur_color = (ct_val / (ct_val + t_val)) * np.array(
                [0, 1, 0]
            ) + (t_val / (ct_val + t_val)) * np.array([1, 0, 0])`

Reply:
Ok I tried that color'ing logic but the plot was super ugly lol. I decided to change it to the following (which gets similar plots as the original and the code is much more readable). I'll add some documentation in this area explaining how the coloring is done.

`
ct_val = max(ct_tiles[tile], default=0)
t_val = max(t_tiles[tile], default=0)

# ct_val and t_val are both guaranteed to fall in [0, 1]
...
cur_color = ct_val * np.array([0, 1, 0]) + t_val * np.array([1, 0, 0])
`